### PR TITLE
TS-5085 `posix_fadvise` is incorrectly used in traffic_logcat

### PR DESCRIPTION
**Pull request by danobi**
`traffic_logcat` was not using `posix_fadvise` correctly. This
patch corrects the original poor usage as well as adds some more
`posix_fadvise` calls that should help with performance.

### CODE REVIEW

**Comment by jpeach** [comment]
[approve ci]

File: proxy/logcat.cc [hunk 1]
```diff
@@ -300,6 +307,9 @@ main(int /* argc ATS_UNUSED */, const char *argv[])
           }
         }
       }
+      // Now that we're done reading a potentially large log file, we can tell the kernel that it's OK to evict
+      // the associated log file pages from cache
+      posix_fadvise(in_fd, 0, 0, POSIX_FADV_DONTNEED);
```
Comment (danobi) [inline proxy/logcat.cc]: Oops, I forgot to wrap this call in `#if`'s. Will fix soon.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1255/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1150/ for details.

**Comment by shinrich** [comment]
Looks good to me.  Can you squash?

**Comment by danobi** [comment]
@shinrich Done.

**Pull request merged**